app/src/main/java/com/github/sdp_begreen/begreen/adapters/FollowingArrayAdapter.kt

```diff
@@ -0,0 +1,89 @@
+package com.github.sdp_begreen.begreen.adapters
+
+import android.content.Context
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.ArrayAdapter
+import android.widget.TextView
+import androidx.annotation.LayoutRes
+import androidx.core.content.ContextCompat
+import androidx.lifecycle.LifecycleCoroutineScope
+import com.github.sdp_begreen.begreen.R
+import com.github.sdp_begreen.begreen.firebase.Auth
+import com.github.sdp_begreen.begreen.firebase.DB
+import com.github.sdp_begreen.begreen.models.User
+import com.google.android.material.button.MaterialButton
+import kotlinx.coroutines.launch
+
+// This adapter is responsible for displaying the users that matches a string written by the User
+// This adapter also lets User follow other users by clicking the follow button
+class FollowingArrayAdapter(context: Context,
+                            @LayoutRes private val resource: Int,
+                            private val users: List<User>,
+                            private val db: DB,
+                            private val auth: Auth,
+                            private val lifeCycle: LifecycleCoroutineScope,
+                            following: List<Boolean>)
+    : ArrayAdapter<User>(context, resource, users) {
+
+    // The boolean at index i is "true" <=> the current user follows user at index i in [users]
+    private val followingUsers = following.toMutableList()
+
+    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
+        return createViewFromResource(position, convertView, parent)
+    }
+
+    override fun getDropDownView(position: Int, convertView: View?, parent: ViewGroup): View {
+        return createViewFromResource(position, convertView, parent)
+    }
+
+    private fun createViewFromResource(position: Int, convertView: View?, parent: ViewGroup): View{
+
+        val view = convertView ?: LayoutInflater.from(context).inflate(R.layout.user_search_bar, parent, false)
+        val textView = view.findViewById<TextView>(R.id.item_text)
+        val user = getItem(position)

```

Comment:
Does the getItem() could return null ? If it is the case, consider adding null checks or using the null-safe ?. operator to prevent potential crashes

Reply:
`getItem` can indeed return `null`, I added a check as you suggested, thanks !